snap/snapcraft.yaml

```diff
@@ -44,19 +44,64 @@ layout:
     bind: $SNAP_COMMON/mnt
 
 parts:
+  libgpiod:
+    # NOTE: The deb package libgpiod2 is not being used anymore because
+    # the ARM64 Runner on GH needs the library in it's API v2 form. 
+    # Refer to https://github.com/canonical/matter-pi-gpio-commander/issues/53
+    plugin: nil
+    source: git://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git
+    source-branch: v2.1
+    override-build: |
+
+      mkdir -p $CRAFT_PART_INSTALL/usr/
+      ./autogen.sh --prefix=$CRAFT_PART_INSTALL/usr --enable-bindings-cxx
+      make install -j$(nproc)
+
+      cp include/gpiod.h $CRAFT_STAGE/gpiod.h
+      
+      mkdir -p /usr/lib/aarch64-linux-gnu
+      cd $CRAFT_PART_INSTALL/usr/lib
+      rm *.a *.la
+
+      if [[ $SNAP_ARCH == "arm64" ]]; then
+        cp $CRAFT_PART_INSTALL/usr/lib/libgpiod.so* /usr/lib/aarch64-linux-gnu
+      fi
+
+      if [[ $SNAP_ARCH == "amd64" ]]; then
+        cp $CRAFT_PART_INSTALL/usr/lib/libgpiod.so* /usr/lib/x86_64-linux-gnu
+      fi
+
+      # Install in the system for all parts
+      cp $CRAFT_PART_INSTALL/usr/include/gpiod.* /usr/include/
```

Comment:
These files are being copied to the build environment and would only exist during the build. Making changes outside the snap's directories is not a recommended practice because the files persist even if the part is removed or changed. Snapcraft won't remove them when the part's cached files are removed.

Files that are under `CRAFT_PART_INSTALL`, get staged and primed by snapcraft. They would end up under the `SNAP` directory of the final artifact, i.e. `/snap/<name>/current/`

Reply:
Understood! It's just that I needed those installed as system libraries for being used during the build.